### PR TITLE
Move CLI commands to a "cli" module

### PR DESCRIPTION
**Pull request by brianhelba**
* Move CLI commands to a "cli" module
* Deprecate legacy "python -m ..." methods for starting Girder
  * The CLI console scripts should be used instead.
* Don't use warnings.warn for runtime server code  …
  * The logger should be used directly, to ensure the output is available for all log handlers.
* Document a coverage exclusion policy for Python code

### CODE REVIEW

File: clients/python/girder_client/__main__.py [hunk 1]
```diff
@@ -17,7 +17,9 @@
...
 ###############################################################################
 
-from girder_client import cli  # pragma: no cover
+import warnings
+from girder_client import cli
 
-
-cli.main()  # pragma: no cover
+if __name__ == '__main__':
+    warnings.warn('Deprecation notice: Use "girder-client" to run the CLI.', DeprecationWarning)
```
Comment (jbeezley) [inline clients/python/girder_client/__main__.py]: I tried running `python -m girder_client`, and I'm not seeing this warning.  Maybe I have something in my environment disabling it...

File: girder/__main__.py [hunk 2]
```diff
@@ -17,42 +17,23 @@
...
-    server.setup(testing)
-
-    cherrypy.engine.start()
-    cherrypy.engine.block()
-
-
-if __name__ == '__main__':  # pragma: no cover
-    main()
+if __name__ == '__main__':
+    girder.logger.warning('Deprecation notice: Use "girder serve" to start Girder.')
```
Comment (jbeezley) [inline girder/__main__.py]: I see the warning here in `~/.girder/logs/error.log` but not on the terminal.  Is that expected?

File: clients/python/girder_client/__main__.py (hunk 1 above)
Comment (brianhelba) [reply clients/python/girder_client/__main__.py]: Fixed to always print.

File: girder/__main__.py (hunk 2 above)
Comment (brianhelba) [reply girder/__main__.py]: Fixed.

File: clients/python/girder_client/__main__.py [hunk 3]
```diff
@@ -17,9 +17,9 @@
 #  limitations under the License.
 ###############################################################################
 
-import warnings
+import six
 from girder_client import cli
 
 if __name__ == '__main__':
-    warnings.warn('Deprecation notice: Use "girder-client" to run the CLI.', DeprecationWarning)
+    six.print_('Deprecation notice: Use "girder-client" to run the CLI.')
```
[the remark this answers is not in the document]
Comment (brianhelba) [reply clients/python/girder_client/__main__.py]: Done. I was trying to be consistent with the behavior of `girder.logprint`, but I agree that `stderr` is more appropriate.

**Comment by brianhelba** [comment]
Thanks to a hint from @danlamanna , I've switched the girder_client warnings to use `click.echo` to `stderr`. The server will continue to use `girder.logprint` (which does go to `stdout`), to ensure that it's goes to the logger, which is important when the server is run as a daemon.

@jbeezley Can you re-approve?

**Pull request merged**